main.go

```diff
@@ -0,0 +1,219 @@
+}
+func ReadFeedback() (feedbacks []Feedbackstruct, err error) {
+	f, err := os.Open("feedback.json")
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	r := bufio.NewReader(f)
+	if err = json.NewDecoder(r).Decode(&feedbacks); err != nil {
+		return nil, err
+	}
+	return feedbacks, nil
+}
+func List(phonebookFile string) {
+	var entries []Entry
+	entries, _ = ReadPhonebook(phonebookFile)
+	Sort(entries)
+	Header()
+	for i, j := range entries {
+		fmt.Printf("%10d %20s %20s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
+		if (i+1)%20 == 0 && i < len(entries) {
+			fmt.Println(" ")
+			fmt.Println("Press <ENTER> to continue...")
+			fmt.Println(" ")
+			fmt.Scanln()
+			Header()
+		}
+	}
+}
+
+func GiveID(entries []Entry) (i uint) {
+	/*for i, j := range entries { // i dont know how to combine it, also in this for i always is 1
+		j.ID = uint(i) + 1
+	}
+	return uint(i) + 1*/
+	i = uint(len(entries)) + 1
+	return i
+}
+
+func Add(scanner *bufio.Scanner, phonebookFile string) {
+	// how to encode with user input??? Tutorial by me (version 1)
+	var entries []Entry                       // make variable where will be all data
+	entries, _ = ReadPhonebook(phonebookFile) // open and encode file with already been information and put it in variable you made in previous step
+
+	var newentries Entry // make new variable with type of your struct, but be careful - it need to be NOT SLICE OF STRUCT, JUST STRUCT, CAUSE WE MAKING ONLY ONE NEW CONTACT, NOT 5 IN ONE TIME
+
+	fmt.Println("Enter last name:") // Scan information you need
+	scanner.Scan()                  // better not use bufio reader, it adds \r\n to text and you ll need 2 variables for string and error
+	scanner.Scan()
+	newentries.LastName = scanner.Text()
+	fmt.Println("Enter first name:")
+	scanner.Scan()
+	newentries.FirstName = scanner.Text()
+	fmt.Println("Enter phone number:")
+	scanner.Scan()
+	newentries.PhoneNumber = scanner.Text()
+	newentries.ID = GiveID(entries)
+
+	entries = append(entries, newentries) // add scanned data to decoded earlier data
+
+	f, err := os.Create("phonebook.json") // and encode it back
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	w := bufio.NewWriter(f)
+	defer w.Flush()
+	if err := json.NewEncoder(w).Encode(entries); err != nil {
+		log.Fatal(err)
+	}
+
+}
+
+func Remove(phonebookFile string) {
+	var entries []Entry                       //var
+	entries, _ = ReadPhonebook(phonebookFile) // opening and decoding
+
+	var noid, nosure uint // actions, work with data
+	fmt.Println("Please, enter ID of contact you want to delete:")
+	fmt.Scan(&noid)
+	for i, j := range entries {
+		if j.ID == noid {
+			fmt.Println(`Are you sure? 
+*the company "Ltd Littleunidragon" is not responsible for "accidentally" permanently deleted contacts
+1) Yes
+2) No `)
+			fmt.Scan(&nosure)
+			if nosure == 1 {
+				entries = append(entries[:i], entries[i+1:]...)
+				fmt.Println("Contact with ID ", noid, " was removed")
+				break
```

Comment:
Since you break in any case, it could be done this way:
```
for i, j := ...
  if j.ID == noid {
    ...
    if nosure == 1 {
      ...
    }
    break
  }
}
```